src/groups/mqb/mqbc/mqbc_clusterdata.h

```diff
@@ -238,6 +238,12 @@ class ClusterData {
 
     // MANIPULATORS
 
+    /// Get a modifiable reference to this object's event scheduler.
+    bdlmt::EventScheduler* scheduler();
+
+    /// Get a modifiable reference to this object's buffer factory.
+    bdlbb::BlobBufferFactory* bufferFactory();
+
```

Comment:
Can the return value be `NULL` here? If not, should we even be returning a pointer type? We should specify.

Reply:
Yeah they cannot be `NULL` here. I will add some asserts in the constructor, and change from pointer to reference here.

Reply:
The issue is that `mqbc::ClusterData` itself only has pointers to components such as `bdlmt::EventScheduler` and `bdlbb::BlobBufferFactory` (see [here](https://github.com/kaikulimu/blazingmq/blob/f3d03d74c4d6bb335080b1d4120d58bcaebe5fc3/src/groups/mqb/mqbc/mqbc_clusterdata.h#L155-L159)). Take the example of `bdlbb::BlobBufferFactory`, the actual owner is [`mqba::Application::d_bufferFactory`](https://github.com/bloomberg/blazingmq/blob/b042ca1fede898611beb53b6979b33c51a273644/src/groups/mqb/mqba/mqba_application.h#L135), and it's passed as pointer from `mqba::Application -> mqbblp::ClusterCatalog -> mqbblp::Cluster -> mqbc::ClusterData`. Given that, would it make sense to keep them as pointers?

Reply:
Will also make copy constructor and copy assignment operator private, since this object is not supposed to be copyable.